decisions/0013-migrating-int-to-bigint-for-primary-keys.md

```diff
@@ -0,0 +1,246 @@
+- Use a batch-processing script (e.g. a delayed job) to populate `id_bigint` for existing rows in both the primary table and, if applicable, all foreign key references.
+- Table locks will be avoided by using a batch processing approach.
+- In case the table has a configurable cleanup duration, the backfill job will only process records which are beyond the cleanup duration to reduce the number of records to be processed. 
+- Backfill will be executed outside the migration due to its potentially long runtime.
+- If necessary the backfill will run for multiple weeks to ensure all records are processed.
+
+#### Step 3a - Migration Pre Check
+- Add a `CHECK` constraint to verify that id_bigint is fully populated (`id_bigint == id & id_bigint != NULL`).
+- In case the backfill is not yet complete or the `id_bigint` column is not fully populated the migration exits gracefully and is retried in the next deploy.
+#### Step 3b - Actual Migration
+- Remove the `CHECK` constraint once verified.
+- Drop the primary key constraint on id.
+- If foreign keys exist, drop the corresponding foreign key constraints.
+- Remove the sync triggers.
+- Switch the primary key by renaming `id_bigint` to `id`.
+- If foreign keys exist, rename `id_bigint` to `id` in referencing tables accordingly.
+- Everything is done in a single transaction to ensure consistency.
+
+### Database Specifics
+
+#### PostgreSQL
+The default value of the `id` column could be either a sequence (for older PostgreSQL versions) or an identity column (for newer PostgreSQL versions).
+This depends on the version of PostgreSQL which was used when the table was initially created.
+The migration script needs to handle both cases.
+
+#### MySQL
+Unknown at this point. Further investigation is needed.
+
+### Rollback Mechanism
+The old `id` column is no longer retained, as the `CHECK` constraint ensures correctness during migration.
+If unexpected issues occur, the migration will fail explicitly, requiring intervention.
+If rollback is needed, either backups could be restored or the migration needs to be reverted manually.
+
+### Standardized Approach
+Write reusable scripts for adding `id_bigint`, setting up triggers, backfilling data, and verifying migration readiness.
+These scripts can be reused for other tables in the future.
+
+### Release Strategy
+
+Steps 1-2 will be released as a cf-deployment major release to ensure that the database is prepared for the migration.  
+Steps 3-4 will be released as a subsequent cf-deployment major release to complete the migration.  
+Between these releases there should be a reasonable time to allow the backfill to complete.
+
+For the `events` table there is a default cleanup interval of 31 days. Therefore, for the `events` table the gap between the releases should be around 60 days.
+
+## Consequences
+### Positive Consequences
+
+- Future-proofing the schema for tables with high record counts.
+- Minimal downtime due to the phased approach.
```

Comment:
No downtime instead of minimal downtime. Maybe: Minimal table lock time due to ...